### PR TITLE
Add form submission handler

### PR DESCRIPTION
**Pull request by he-james**
Currently just extracts out the input value and id, can be adjusted however preferred based on the event / target objects 

image

### CODE REVIEW

File: components/stepped-form/index.tsx [hunk 1]
```diff
@@ -168,14 +168,24 @@ interface FormField {
...
-  const [formData, setFormData] = useState<object>({});
+  const [formData, setFormData] = useState<FormData>();
   const [loading, setLoading] = useState<boolean>(true);
 
-  const initForm = () => {
+  const initForm = async () => {
     setLoading(true);
     const getFormData = getFormTemplate("1", "1");
-    setFormData(getFormData);
+    setFormData(await getFormData);
```
Comment (sdmungov) [inline components/stepped-form/index.tsx]: this is not where await would go, I think you would put it in front of getFormTemplate so "= await getFormTemplate"

File: components/stepped-form/index.tsx [hunk 2]
```diff
@@ -184,6 +194,22 @@ const SteppedForm = () => {
     initForm();
   }, []);
 
+  const handleSubmit = async (event) => {
+    event.preventDefault()
+    // array of objects containing id and value from input
+    const responses = []
+    // get id and values from input field
+    for (const response of event.target) {
+      responses.push({id: response.name, value: response.value})
```
Comment (sdmungov) [inline components/stepped-form/index.tsx]: would a .map or .foreach or even a .reduce be better here than a loop, havent seen var loops in js in a while

**Review by sdmungov — CHANGES_REQUESTED** [review]

**Comment by sdmungov** [comment]
@he-james we als need to change how we do id's for the questions that have the same id but need multiple inputs like ranking and checkboxes because rn in the submission it looks like they all share the same value since they share the same id. I'm thinking we do id-index so 15-input for all of them would change to 15-0-input 15-1-input etc.

File: components/stepped-form/index.tsx (hunk 1 above)
Comment (he-james) [reply components/stepped-form/index.tsx]: either works - depends if you want getFormData to explicitly be the response object or a Promise - in this case was just fixing a linter warning

File: components/stepped-form/index.tsx (hunk 2 above)
Comment (he-james) [reply components/stepped-form/index.tsx]: yes - although depends on use case. `for...of` is good for readability and only incrementally slower. had I more brain cells the ideal solution is a basic `.map`

**Comment by sdmungov** [comment]
@he-james I am happy with it once we get that vercel thing fixed, heres the error. The request is being blacked because something to do with http vs https and so it returns undefined as our get response instead of the form data. 
Screen Shot 2021-08-22 at 1 36 09 AM

File: components/stepped-form/index.tsx [hunk 3]
```diff
@@ -159,189 +159,239 @@ const Icon = styled(FontAwesomeIcon)`
...
+        rankingResponse.splice(+index, 1, +value)
+      }
+      responses.push({id: +id, response: rankingResponse})
+    }
+    const request = new SubmitFormDto()
+    request.responses = responses;
+    // Submit!
+    await postFormSubmission('1', '1', {responses})
+    // #TODO - redirect home and try-catch with rendered error message "Something went wrong, try again (err: error_code)"
+    props.history.push('')
```
Comment (sdmungov) [inline components/stepped-form/index.tsx]: Try .push('/')

File: components/stepped-form/index.tsx [hunk 4]
```diff
@@ -159,189 +159,239 @@ const Icon = styled(FontAwesomeIcon)`
...
+        const value = rankingMap[id][index]
+        rankingResponse.splice(+index, 1, +value)
+      }
+      responses.push({id: +id, response: rankingResponse})
+    }
+    const request = new SubmitFormDto()
+    request.responses = responses;
+    // Submit!
+    await postFormSubmission('1', '1', {responses})
+    // #TODO - redirect home and try-catch with rendered error message "Something went wrong, try again (err: error_code)"
```
Comment (sdmungov) [inline components/stepped-form/index.tsx]: In all my react native stuff it's usually
```
catch(error) {
  Alert.message('something went wrong, ${error.message}')
}
```
Or at least something along those linez

**Review by sdmungov — APPROVED** [review]

File: components/stepped-form/index.tsx (hunk 3 above)
Comment (he-james) [reply components/stepped-form/index.tsx]: doesn't work - looks like next.js may have its own shit https://nextjs.org/docs/api-reference/next.config.js/redirects

**Pull request merged**